sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -804,6 +805,7 @@ public TimeseriesQuery toTimeseriesQuery()
     final Map<String, Object> theContext = new HashMap<>();
     theContext.put("skipEmptyBuckets", true);
     theContext.putAll(plannerContext.getQueryContext());
+    theContext.remove(ContextLiteralLookupOperatorConversion.CONTEXT_LITERAL_LOOKUP_PROPERTY);
```

Comment:
This removal from the context is only happening for timeseries, but I guess you meant to have it happen for all query types. Although, if you take the suggestion above to make them top-level, this approach you're using here won't work. I'm not sure what we could do instead. Maybe leave it in the context? Maybe have a convention for context parameter naming that will get removed before sending queries down past the broker?

Reply:
Oops, this was an oversight, I meant to remove from all query types, good catch.

I think we definitely don't want to leave it in the context, since we would be significantly inflating the size of query json since all of those values would now be duplicated in one way or another as part of the native Druid query.

Naming conventions make me uncomfortable and don't seem like an awesome experience, would it be possible to build some sort of map of encountered replacement variables as part of parsing the sql expression into a Druid query to then erase them from the context? I'll poke around and see if this could be a thing somehow since it seems like it would probably be best.